proxy/modules/native_module_host/src/native_module_host.c

```diff
@@ -0,0 +1,418 @@
+		else
+		{
+			MODULE_LOADER_API* loader_api = gw_loader_info->loader->api;
+			/*Codes_SRS_NATIVEMODULEHOST_17_020: [ NativeModuleHost_Create shall get the MODULE_API pointer from the loader. ]*/
+			const MODULE_API* module_apis = loader_api->GetApi(gw_loader_info->loader, module_host->module_library_handle);
+			if (module_apis == NULL)
+			{
+				/*Codes_SRS_NATIVEMODULEHOST_17_026: [ If any step above fails, then NativeModuleHost_Create shall free all resources allocated and return NULL. ]*/
+				loader_api->Unload(gw_loader_info->loader, module_host->module_library_handle);
+				result = __LINE__;
+			}
+			else
+			{
+				/*Codes_SRS_NATIVEMODULEHOST_17_021: [ NativeModuleHost_Create shall call the module's _ParseConfigurationFromJson on the "module.args" object. ]*/
+				const void* module_configuration = MODULE_PARSE_CONFIGURATION_FROM_JSON(module_apis)(remote_module_args);
+				/*Codes_SRS_NATIVEMODULEHOST_17_022: [ NativeModuleHost_Create shall build the module confguration from the parsed entrypoint and parsed module arguments. ]*/
+				const void* transformed_module_configuration = loader_api->BuildModuleConfiguration(
+					gw_loader_info->loader,
+					gw_loader_info->entrypoint,
+					module_configuration
+				);
+				/*Codes_SRS_NATIVEMODULEHOST_17_023: [ NativeModuleHost_Create shall call the module's _Create on the built module configuration. ]*/
+				module_host->module = MODULE_CREATE(module_apis)(broker, transformed_module_configuration);
+				if (module_host->module == NULL)
+				{
+					/*Codes_SRS_NATIVEMODULEHOST_17_026: [ If any step above fails, then NativeModuleHost_Create shall free all resources allocated and return NULL. ]*/
+					loader_api->Unload(gw_loader_info->loader, module_host->module_library_handle);
+					result = __LINE__;
+				}
+				else
+				{
+					/*Codes_SRS_NATIVEMODULEHOST_17_025: [ NativeModuleHost_Create shall return a non-null pointer to a MODULE_HANDLE on success. ]*/
+					module_host->module_loader = gw_loader_info->loader;
+					module_host->module_host_broker = broker;
+						
+					result = 0;
+				}
+				/*Codes_SRS_NATIVEMODULEHOST_17_024: [ NativeModuleHost_Create shall free all resources used during module loading. ]*/
+				MODULE_FREE_CONFIGURATION(module_apis)((void*)module_configuration);
+				loader_api->FreeModuleConfiguration(gw_loader_info->loader, transformed_module_configuration);
+			}
+		}
+	}
+	return result;
+}
+
+// We are going to receive a string in the configuration, It should be a JSON string with 2 required fields:
+// "outprocess.loader" and "module.arguments" will return a pointer to "MODULE_HOST" data structure.
+// This will:
+//	1. Initialize the module loader.
+//	2. Parse the JSON string
+//	3. Extract the loader args and module args
+//	4. Get the loader from the loader args.
+//	5. Use the loader to get the module API
+//	6. Create the module using the module args.
+//	7. Save all of this stuff in the MODULE_HOST and return that as the remote_module.
```

Comment:
We already have requirement document. Adding a new point of documenting behavior will easily get out dated.... If/when we need to change code we will have to change this header...